cpp/src/io/orc/writer_impl.cu

```diff
@@ -314,10 +314,33 @@ class orc_column_view {
   std::optional<uint32_t> _parent_index;
 };
 
-size_type orc_table_view::num_rows() const noexcept
-{
-  return columns.empty() ? 0 : columns.front().size();
-}
+/**
+ * Non-owning view of a cuDF table that includes ORC-related information.
+ *
+ * Columns hierarchy is flattened and stored in pre-order.
+ */
+struct orc_table_view {
```

Comment:
Also not sure why, but the 17 compiler was fine with forward-declared `orc_column_view`, while 20 is not.